### PR TITLE
Jeff Schock

### PR DESCRIPTION
**Pull request by etisdew**
Took me a while to find a time function that didnt have a massive footprint, I hope this is accurate enough for you.

### CODE REVIEW

File: ex1/ex1.c [hunk 1]
```diff
@@ -5,10 +5,25 @@
...
+#include <sys/types.h>
+#include <sys/wait.h>
 
 int main(int argc, char *argv[])
 {
-    // Your code here
+    pid_t FPID = fork();
+    pid_t PID = getpid();
+    printf("Starting... %d", PID);
+    pid_t PPID = getppid();
```
Comment (adamlower) [inline ex1/ex1.c]: You should initialize a variable here and then change it in the child and parent processes.

File: ex3/ex3.c [hunk 2]
```diff
@@ -5,10 +5,34 @@
...
+int main(int argc, char *argv[])
 {
-    // Your code here
+    pid_t FPID = fork();
+    pid_t PID = getpid();
+    printf("Starting... %d", PID);
+    pid_t PPID = getppid();
+
+    FILE *fp;
+    char file[] = "text.txt";
```
Comment (adamlower) [inline ex3/ex3.c]: File does not exist

File: ex3/ex3.c [hunk 3]
```diff
@@ -5,10 +5,34 @@
...
+    fp = fopen(file, "a");
+
+    if (fp != NULL)
+    {
+        if (FPID != 0)
+        {
+            wait(NULL);
+            fprintf(fp, "Goodbye\n", PID, PPID, FPID);
```
Comment (adamlower) [inline ex3/ex3.c]: Too many arguments for the given string

File: ex3/ex3.c [hunk 4]
```diff
@@ -5,10 +5,34 @@
...
+            fclose(fp);
+        }
+        if (FPID == 0)
+        {
+            fprintf(fp, "Hello\n", PID, PPID, FPID);
```
Comment (adamlower) [inline ex3/ex3.c]: Too many arguments for the given string

File: ex3/ex3.c (hunk 3 above)
Comment (adamlower) [reply ex3/ex3.c]: Could just be `printf("Goodbye\n");`

File: ex3/ex3.c (hunk 4 above)
Comment (adamlower) [reply ex3/ex3.c]: Could just be `printf("Hello\n");`

File: ex1/ex1.c (hunk 1 above)
Comment (etisdew) [reply ex1/ex1.c]: I have updated to reflect this request.

**Comment by adamlower** [comment]
Hey Jeff, great job on this sprint! Your code was pretty clean and consistent throughout. I only really noticed a few errors throughout and everything seemed to work pretty well. Your ex6 solution seems to work well, even without the clock_gettime function. Keep up the good work!

File: ex3/ex3.c (hunk 3 above)
Comment (etisdew) [reply ex3/ex3.c]: I was told to just use this as a scratch pad to learn the api. Noone mentioned any limits to testing args.

File: ex3/ex3.c (hunk 2 above)
Comment (etisdew) [reply ex3/ex3.c]: Very sorry for the confusion, the argument "a" used in the subsequent fopen(...) call allows the .txt to be generated on an as needed basis overwriting the previous output for clarity.

File: ex3/ex3.c (hunk 3 above)
Comment (adamlower) [reply ex3/ex3.c]: Absolutely! You're free to use as many arguments as you'd like, but you need to provide the % format specifiers for each argument.

File: ex3/ex3.c (hunk 2 above)
Comment (adamlower) [reply ex3/ex3.c]: Ah, you're right! My system wasn't allowing the file to be made with the warnings from fprintf, but it does work as expected :)

**Pull request closed without merging**